devtools/x-core/src/lib.rs

```diff
@@ -29,7 +29,7 @@ pub struct XCoreContext {
     config: XCoreConfig,
     current_dir: PathBuf,
     current_rel_dir: PathBuf,
-    git_cli: GitCli,
+    git_cli: Option<GitCli>,
```

Comment:
This seemed the best way to communicate via the struct we may not have git support.   also, even though gitcli types are private to the x-core crate, they bleed out via this public reference in to linters --- it seems it might be better to move the determinator logic in to xcore and not even show gitcli?  and determinator would be a no op in the case of no git?

Reply:
I think making the git cli public is fine.